### DISCUSSION TITLE
Using GeckoView instead of WebView

### DISCUSSION THREAD

**Discussion by Ranomez** [plateaukao/einkbro#307] [Ideas]
I've been using Android based e-book readers for a long time and while some offer Google services many don't and as such don't allow using a different WebView than the built-in one (even after rooting and installing GAPPS changing WebView isn't easy) which is always severely outdated breaking sites. The situation on devices that support Google services isn't necessarily better either as eventually newer WebView versions won't support older versions of Android and I've not yet had an e-book reader that got a single Android OS version upgrade.
There's an alternative to WebView called GeckoView, by Mozilla, which can be built into the app and will provide a consistent experience across devices regardless of Android version and support for switching the used WebView implementation.
I've already seen a few other browsers switch to it so I'm wondering if that option has been considered here since I saw quite a few issues submitted where the real problem was the outdated WebView?
I'm not saying necessarily replacing WebView entirely but maybe a separate build or an option to switch to GeckoView in the app (TV Bro has that currently but it's temporary there from what I gather)?

**Comment by plateaukao** [comment]
EinkBro is Open Source. Feel free to fork it and integrate with any other WebView engines, and share it with others who have the same requirement.
